### DISCUSSION TITLE
Configuring ObjectMapper for GraphQLResponse extractor

### DISCUSSION THREAD

**Discussion by onomesotu** [Netflix/dgs-framework#1042] [General]
I am testing DGS mutation in the web layer via CustomGraphQLClient and GraphQLQueryRequest builder like so: 
```
final GraphQLQueryRequest graphQLQueryRequest = new GraphQLQueryRequest(
        someMutation.newRequest()
            .input(
                SomeMutationInput.newBuilder()
                    .id("some-id")
                    .build()
            )
            .build(),
        new ResponseResult().onOneType().id()
    );
```
Note that all my types are generated using the DGS codegen. 

In my test, I am simply extracting response and mapping to my GQL type like so:
final var myTpe = graphQLResponse.extractValueAsObject(JSON_PATH, MyType.class);

I get an exception because the server correctly returns a type name  of the resource because its a union type

```
java.lang.IllegalArgumentException: Unrecognized field "__typename" (class com.package.generated.graphql.types.MyType), not marked as ignorable (one known property: "id"])
 at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: com.package.generated.graphql.types.MyType["__typename"])
```

This is simply an issue with the ObjectMapper failing due to an unknown field. 

Is there a way to configure this Object mapper? 
I have tried different approaches including using what's described in the doc for overriding the object mapper but it seems it doesn't use that. 

Internally, I see this is how the object mapper used is configured.
```
private val mapper: ObjectMapper = jacksonObjectMapper()
            .registerModule(JavaTimeModule())
            .registerModule(ParameterNamesModule())
            .registerModule(Jdk8Module())
            .enable(DeserializationFeature.READ_UNKNOWN_ENUM_VALUES_USING_DEFAULT_VALUE)
```

I can deserialize with my own ObjectMapper but it will be nice to be able to set the configuration of the object mapper used here `graphQLResponse.extractValueAsObject`

Thoughts?

**Comment by srinivasankavitha** [comment]
Yes, I believe this has come up in another issue as well. It does make sense to be able to configure the object mapper used to extract the response. We will add this in an upcoming release. Thanks for the detailed description.

**Reply by srinivasankavitha** [reply]
https://github.com/Netflix/dgs-framework/issues/1021

**Comment by onomesotu** [comment]
Thanks, @srinivasankavitha. Looking forward to it!

**Comment by Rfank2021** [comment]
I also met the problem and I found the __typename__ field is added by apollo client so we can config the behavior, https://community.apollographql.com/t/how-to-remove-typename-for-a-specific-query/1416
